### PR TITLE
Force permissions on windows capture as well

### PR DESCRIPTION
**Pull request by florin0x01**

### CODE REVIEW

File: plugins/mac-capture/mac-window-capture.m [hunk 1]
```diff
@@ -102,6 +104,13 @@ static inline void capture_frame(struct window_capture *wc)
 
 	da_init(wc->buffer);
 
+	blog(LOG_INFO, "[window-capture] - Init Display Capture for permissions dialog");
+	
+	void *dc = display_capture_create(settings, source);
```
Comment (florin0x01) [inline plugins/mac-capture/mac-window-capture.m]: Makes call to https://developer.apple.com/documentation/coregraphics/1454968-cgdisplaystreamcreatewithdispatc?language=objc which also pops up the permissions screen recording dialog

Comment (summeroff) [reply plugins/mac-capture/mac-window-capture.m]: do we need a "dc" later or can release it here?

Comment (florin0x01) [reply plugins/mac-capture/mac-window-capture.m]: no, only for null check.

Comment (summeroff) [reply plugins/mac-capture/mac-window-capture.m]: then display_capture_destroy to not leak memory. 

is there a way to check for permission without creating and initializing display capture?

Comment (florin0x01) [reply plugins/mac-capture/mac-window-capture.m]: not that I know of.

Comment (summeroff) [reply plugins/mac-capture/mac-window-capture.m]: yes. looks like it is a way of macos permissions. you need to create display object to trigger permission request. 
maybe not reuse display_capture_create but create new function what only call CGDisplayStreamCreate with minimum parameters and release.

Comment (florin0x01) [reply plugins/mac-capture/mac-window-capture.m]: @summeroff Reordered the code a bit more to have less exposure to that display_capture_create function, retested it and it seems to work. Of course if also Trevor can retest that would be best even.

**Comment by florin0x01** [comment]
@EddyGharbi did some rewrite here to separate a bit the structures.

File: plugins/mac-capture/CMakeLists.txt [hunk 2]
```diff
@@ -14,14 +14,16 @@ include_directories(${COREAUDIO}
 
 set(mac-capture_HEADERS
 	audio-device-enum.h
-	window-utils.h)
+	window-utils.h
+    screen-utils.h)
```
Comment (EddyGharbi) [inline plugins/mac-capture/CMakeLists.txt]: style issue here, looks like 2 spaces are missing

File: plugins/mac-capture/CMakeLists.txt [hunk 3]
```diff
@@ -14,14 +14,16 @@ include_directories(${COREAUDIO}
...
 
 set(mac-capture_SOURCES
 	plugin-main.c
 	audio-device-enum.c
 	mac-audio.c
 	mac-display-capture.m
 	mac-window-capture.m
+    screen-utils.m
```
Comment (EddyGharbi) [inline plugins/mac-capture/CMakeLists.txt]: same style issue here

File: plugins/mac-capture/mac-window-capture.m [hunk 4]
```diff
@@ -25,7 +22,7 @@
 	os_event_t *capture_event;
 	os_event_t *stop_event;
 };
-
+
```
Comment (EddyGharbi) [inline plugins/mac-capture/mac-window-capture.m]: looks like a non intended extra space added

File: plugins/mac-capture/mac-window-capture.m [hunk 5]
```diff
@@ -102,6 +99,22 @@ static inline void capture_frame(struct window_capture *wc)
...
+        return NULL;       
+    }
+    dc->display = obs_data_get_int(settings, "display");
+
+    if (!init_screen_stream(dc)) {
+        blog(LOG_INFO, "[window-capture] - Display Capture Init Fail");
+        bfree(dc);
+        return NULL;
+    }
+    bfree(dc);
```
Comment (EddyGharbi) [inline plugins/mac-capture/mac-window-capture.m]: wrong indentation for this block

File: plugins/mac-capture/mac-window-capture.m [hunk 6]
```diff
@@ -102,6 +99,22 @@ static inline void capture_frame(struct window_capture *wc)
...
+	blog(LOG_INFO, "[window-capture] - Init Display Capture for permissions dialog");
+	
+    struct screen_capture *dc = bzalloc(sizeof(struct screen_capture));
+    if (!dc) {
+        blog(LOG_INFO, "[window-capture] - Display Capture Alloc Fail"); 
+        return NULL;       
+    }
+    dc->display = obs_data_get_int(settings, "display");
+
+    if (!init_screen_stream(dc)) {
```
Comment (EddyGharbi) [inline plugins/mac-capture/mac-window-capture.m]: what does this implies? We are only using this function for the window capture in order to ask for the permission but we don't want to capture the screen. Should we close the screen stream when this is done?

Comment (EddyGharbi) [reply plugins/mac-capture/mac-window-capture.m]: I see a `bfree(dc)` below, are we counting on that to close everything that needs to be closed?

Comment (florin0x01) [reply plugins/mac-capture/mac-window-capture.m]: yes

File: plugins/mac-capture/mac-window-capture.m (hunk 5 above)
Comment (florin0x01) [reply plugins/mac-capture/mac-window-capture.m]: appears fine for me

Comment (EddyGharbi) [reply plugins/mac-capture/mac-window-capture.m]: could you check again, I still still the indentation issue

Comment (EddyGharbi) [reply plugins/mac-capture/mac-window-capture.m]: @florin0x01 ^

File: plugins/mac-capture/mac-window-capture.m (hunk 1 above)
Comment (EddyGharbi) [reply plugins/mac-capture/mac-window-capture.m]: @summeroff could you update this conversation: should it be marked as resolved or do you still have some comments?

**Review by EddyGharbi — APPROVED** [review]

**Pull request merged**